shared/podman/images.go

```
@@ -6,48 +6,170 @@ package podman
+		for _, metadataTag := range data.Image.Tags {
+			if metadataTag == tag {
+				return fullPathFile, nil
+			}
+		}
+	}
+	return "", nil
+}
+
+// GetRpmImagePath return the RPM image path.
+func GetRpmImagePath(image string) (string, error) {
+	log.Debug().Msgf("Looking for installed RPM package containing %s image", image)
+
+	rpmImageFile, tag := GetRpmImageName(image)
 
+	files, err := os.ReadDir(rpmImageDir)
+	if err != nil {
+		return "", fmt.Errorf("cannot read directory %s: %s", rpmImageDir, err)
+	}
+
+	for _, file := range files {
+		if !strings.HasSuffix(file.Name(), "metadata") {
+			continue
+		}
+		fullPathFileName := path.Join(rpmImageDir, file.Name())
+		log.Debug().Msgf("Parsing metadata file %s", fullPathFileName)
+		fileHandler, err := os.Open(fullPathFileName)
+		if err != nil {
+			log.Debug().Msgf("Error opening metadata file %s: %s", fullPathFileName, err)
+			continue
+		}
+		defer fileHandler.Close()
+		byteValue, err := io.ReadAll(fileHandler)
+		if err != nil {
+			log.Debug().Msgf("Error reading metadata file %s: %s", fullPathFileName, err)
+			continue
+		}
+
+		fullPathFile, err := BuildRpmImagePath(byteValue, rpmImageFile, tag)
+		if err != nil {
+			log.Warn().Msgf("Cannot unmarshal metadata file %s: %s", fullPathFileName, err)
+			return "", err
+		}
+		if len(fullPathFile) > 0 {
+			log.Debug().Msgf("%s match with %s", fullPathFileName, image)
+			return fullPathFile, nil
+		}
+		log.Debug().Msgf("%s does not match with %s", fullPathFileName, image)
+	}
+	log.Debug().Msgf("No installed RPM package containing %s image", image)
+	return "", fmt.Errorf("image %s does not exists as RPM image", image)
+}
+
+func loadRpmImage(rpmImageBasePath string) (string, error) {
+	out, err := utils.RunCmdOutput(zerolog.DebugLevel, "podman", "load", "--quiet", "--input", rpmImageBasePath)
+	if err != nil {
+		return "", fmt.Errorf("cannot load image from: %s, continuing trying to pull from the registry: %s", rpmImageBasePath, err)
```

This message will be concatenated with another and look like this:
```
Cannot use RPM image for %s: cannot load image from: %s, continuing trying to pull from the registry: %s
```
I would just return the error here:
```suggestion
		return "", err
```